pandas/core/indexing.py

```diff
@@ -888,26 +888,22 @@ def _getitem_nested_tuple(self, tup: tuple):
         # handle the multi-axis by taking sections and reducing
         # this is iterative
         obj = self.obj
-        axis = 0
-        for key in tup:
+        # GH#41369 Loop in reverse order to avoid dtype conversion when converting df
+        # row to a series
```

Comment:
IIUC the reason this works is bc this ensures we index along columns before rows, so that in multi-block cases we select only necessary blocks, avoiding the fast_xs/interleave/whatever call that is doing the casting ATM?

Reply:
Yes exactly

Reply:
can you flesh out the comment to that effect?

Reply:
pending that, LGTM

Reply:
Done